src/foam/nanos/column/NestedPropertiesExpression.js

```diff
@@ -81,61 +83,124 @@ foam.CLASS({
         {
           name: 'i',
           class: 'Int'
+        },
+        {
+          name: 'expr',
+          javaType: 'foam.mlang.Expr'
         }
       ],
-      code: function (of, propName, i) {
-        if ( i === propName.length - 1 )
-          return of.getAxiomByName(propName[i]);
+      code: function (of, propName, i, expr) {
         var prop = of.getAxiomByName(propName[i]);
-        return foam.mlang.Expressions.create().DOT(prop, this.returnPropExpr(prop, this.returnDotExprForNestedProperty(prop.of, propName, ++i)));
+        if ( ! prop ) return null;
+
+        var propExpr = this.buildPropertyExpr(prop, expr);
+
+        if ( i === propName.length - 1 )
+          return propExpr;
+
+        return this.returnDotExprForNestedProperty(prop.of, propName, ++i, propExpr);
       },
       javaCode: `
-        ClassInfo ci = of;
-        PropertyInfo p = (PropertyInfo) ci.getAxiomByName(propName[i]);
-        if ( i == propName.length - 1 ) {
-          return p;
-        }
+        PropertyInfo prop = (PropertyInfo) of.getAxiomByName(propName[i]);
+
+        if ( prop == null ) return null;
+
+        Method m = getFinderMethod(prop);
+        expr = buildPropertyExpr(prop, expr);
```

Comment:
please create new variable
```suggestion
        var propExpr = buildPropertyExpr(prop, expr);
```

Reply:
fixed.